### PR TITLE
support type alias

### PR DESCRIPTION
**Pull request by zhfnjust**

### CODE REVIEW

File: test/alias.test.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+    it('should success when using all int alias', () => {
+        let result = alias.setToken([new Time(10),new Age(3),new Token(3)]).verify()
+        assert.isTrue(result.success, result.error);
+    })
+
+    // it('should success when using all int alias', () => {
+    //     let result = alias.setToken(new Tokens([new Token(10),new Token(3),new Token(3)])).verify()
+    //     assert.isTrue(result.success, result.error);
+    // })
```
Comment (xhliu) [inline test/alias.test.ts:109]: uncomment?

File: test/alias.test.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    token: new Token(101)
+}));
+
+
+describe('Alias type check', () => {
+
+    it('should success when using MaleAAA', () => {
+
+        let result = alias.unlock(new MaleAAA({
+            name: new Name("68656c6c6f20776f726c6421"),
```
Comment (xhliu) [inline test/alias.test.ts:35]: Also test `new Bytes()` somewhere, here or below.

File: src/utils.ts [hunk 3]
```diff
@@ -626,4 +609,30 @@ export function genLaunchConfigFile(constructorArgs: SupportedParamType[], pubFu
   ), 2));
   return path2uri(file);
 
+}
+
+/***
+ * map alias to real type
+ */
+export function findRealFinalType(alias: AliasEntity[], structs: StructEntity[], type: string): string {
```
Comment (xhliu) [inline src/utils.ts:617]: rename findRealFinalType -> resolveType?

File: src/utils.ts [hunk 4]
```diff
@@ -626,4 +609,30 @@ export function genLaunchConfigFile(constructorArgs: SupportedParamType[], pubFu
   ), 2));
   return path2uri(file);
 
+}
+
+/***
+ * map alias to real type
```
Comment (xhliu) [inline src/utils.ts:615]: resolve type

File: src/contract.ts [hunk 5]
```diff
@@ -326,27 +325,69 @@ export function buildContractClass(desc: CompileResult | ContractDescription): a
...
     });
 
   });
 
   return structTypes;
 }
+
+
+
+export function buildTypeClass(desc: CompileResult | ContractDescription): Record<string, typeof ScryptType> {
```
Comment (xhliu) [inline src/contract.ts:352]: 👍 

buildTypeClass -> buildTypeClass**es**

**Review by xhliu — APPROVED** [review]
Very good 💯 

Just some refactor.

File: test/alias.test.ts (hunk 1 above)
Comment (zhfnjust) [reply test/alias.test.ts]: remove

**Pull request merged**